src/handlers/validate_config.rs

```diff
@@ -0,0 +1,150 @@
+//! For pull requests that have changed the triagebot.toml, validate that the
+//! changes are a valid configuration file.
+//! It won't validate anything unless the PR is open and has changed.
+//!
+//! These are the implementation steps as follows:
+//!   1. If the issue is of type pull request and it is open, then continue.
+//!   2. Fetch the triagebot.toml from the HEAD branch of the PR.
+//!   3. If the triagebot.toml is valid, then continue.
+//!   4. If the triagebot.toml is invalid, then report the error.
+//!   5. If the PR hasn't changed (no new commit been pushed), use cached triagebot.toml content.
+
+use crate::{
+    config::ValidateConfig,
+    github::IssuesAction,
+    handlers::{Context, IssuesEvent},
+};
+
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+use tracing as log;
+
+/// Caching the content of triagebot.toml across commit SHA to save network round-trip
+/// time. The content will look like this:
+/// COMMIT_SHA => CONTENT_OF_TRIAGEBOT_TOML
+static TRIAGEBOT_CACHE: Lazy<Mutex<LifoHashMap<String, Vec<u8>>>> =
+    Lazy::new(|| Mutex::new(LifoHashMap::new(128)));
+
+mod lifo;
+use lifo::LifoHashMap;
+
+/// If the issue is a pull request and if among the changed files, triagebot.toml
+/// is changed, then validate the triagebot.toml config file with the syntax
+/// parser from the config module.
+pub(super) async fn parse_input(
+    ctx: &Context,
+    event: &IssuesEvent,
+    _config: Option<&ValidateConfig>,
+) -> Result<Option<()>, String> {
+    if !event.issue.is_pr() {
+        log::debug!("Ignoring issue: {:?}", event.issue);
+        return Ok(None);
+    }
+
+    if !matches!(
+        event.action,
+        IssuesAction::Opened | IssuesAction::Synchronize | IssuesAction::ReadyForReview
+    ) {
+        log::debug!("Ignoring issue action: {:?}", event.action);
+        return Ok(None);
+    }
+
+    log::debug!("Validating triagebot config");
+    match validate_triagebot_config(ctx, event).await {
+        Ok(()) => {
+            log::debug!("Valid triagebot config");
+            Ok(Some(()))
+        }
+        Err(e) => {
+            log::error!("Invalid triagebot config: {:?}", e);
+            Err(e)
+        }
+    }
+}
+
+async fn get_triagebot_content(repo: &str, git_ref: &str, sha: &str) -> Vec<u8> {
+    // Try to hit the cache first
+    if let Some(triagebot_toml) = TRIAGEBOT_CACHE.lock().unwrap().get(&sha.to_string()) {
+        log::info!(
+            "Cache hit for triagebot.toml for repository {} and commit {}",
+            repo,
+            sha
+        );
+        return triagebot_toml.clone();
+    }
+
+    log::warn!(
```

Comment:
I think doing a warn! on every cache miss would pollute the logs with unactionable information. I would expect something like this to be trace or debug.  Similarly with the info! above about a hit, that information is likely to add noise.